compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -463,16 +463,31 @@ pub(crate) struct TyS<'tcx> {
 #[rustc_pass_by_value]
 pub struct Ty<'tcx>(Interned<'tcx, WithStableHash<TyS<'tcx>>>);
 
-// Statics only used for internal testing.
-pub static BOOL_TY: Ty<'static> = Ty(Interned::new_unchecked(&WithStableHash {
-    internee: BOOL_TYS,
-    stable_hash: Fingerprint::ZERO,
-}));
-const BOOL_TYS: TyS<'static> = TyS {
-    kind: ty::Bool,
-    flags: TypeFlags::empty(),
-    outer_exclusive_binder: DebruijnIndex::from_usize(0),
-};
+const LEAKED_BOOL_TY_ALREADY: std::sync::atomic::AtomicBool =
+    std::sync::atomic::AtomicBool::new(false);
+
+/// "Static" bool only used for internal testing.
+///
+/// FIXME(rustc_type_ir): This really should be replaced with something that doesn't leak.
+/// however, since it's used for testing, it's not _that_ bad.
+pub fn leak_bool_ty_for_unit_testing<'tcx>() -> Ty<'tcx> {
```

Comment:
Two alternatives I can think of (not mutually exclusive):
1) Move this closer to the use site.
2) If the test is under `cfg(test)`, then put this under `cfg(test)` too.

Reply:
I can't put this closer to the use site, because constructing an `Interned` requires a private constructor that's only accessible from middle, and the usage is in a different crate.

I can't put it under `cfg(test)` for the same reason -- `cfg(test)` doesn't propagate, right?